### PR TITLE
cardano-api-internal: ProposeNewCommittee: StakeKey -> CommitteeColdKey

### PR DESCRIPTION
**Pull request by smelc**
# Changelog

```yaml
- description: |
    Make ProposeNewCommittee use the appropriate type of key
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Required to implement [this comment](https://github.com/input-output-hk/cardano-cli/issues/300#issuecomment-1731004496) of https://github.com/input-output-hk/cardano-cli/issues/300.

# Checklist

- [X] Commit sequence broadly makes sense and commits have useful messages
- [X] The change log section in the PR description has been filled in
- NA New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- NA The version bounds in `.cabal` files are updated
- [x] CI passes. See note on CI.  The following CI checks are required:
  - [x] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [x] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [X] The changelog section in the PR is updated to describe the change
- [X] Self-reviewed the diff

### CODE REVIEW

File: cardano-api/internal/Cardano/Api/Governance/Actions/ProposalProcedure.hs [hunk 1]
```diff
@@ -224,12 +224,12 @@ createAnchor url anchorData =
 -- TODO conversions that likely need to live elsewhere and may even deserve
 -- additional wrapper types
 
-toCommitteeMember :: Hash StakeKey -> L.Credential ColdCommitteeRole StandardCrypto
-toCommitteeMember (StakeKeyHash keyhash) = coerceKeyRole $ L.KeyHashObj keyhash
+toCommitteeMember :: Hash CommitteeColdKey -> L.Credential ColdCommitteeRole StandardCrypto
```
Comment (Jimbo4350) [inline cardano-api/internal/Cardano/Api/Governance/Actions/ProposalProcedure.hs:227]: So what we need to do is remove `toCommitteeMember` and `fromCommitteeMember`. Previously we were coercing the `StakeKey` key role to  `ColdCommitteeRole`. Now this is not necessary so we can delete these functions.

**Review by Jimbo4350 — APPROVED** [review]
LGTM! One comment.

File: cardano-api/internal/Cardano/Api/Governance/Actions/ProposalProcedure.hs (hunk 1 above)
Comment (smelc) [reply cardano-api/internal/Cardano/Api/Governance/Actions/ProposalProcedure.hs:227]: Done :+1:

File: cardano-api/internal/Cardano/Api/Governance/Actions/ProposalProcedure.hs [hunk 2]
```diff
@@ -224,12 +224,12 @@ createAnchor url anchorData =
...
+toCommitteeMember (CommitteeColdKeyHash keyhash) = L.KeyHashObj keyhash
 
-fromCommitteeMember :: L.Credential ColdCommitteeRole StandardCrypto -> Hash StakeKey
-fromCommitteeMember = (. coerceKeyRole) $ \case
-  L.KeyHashObj keyhash -> StakeKeyHash keyhash
+fromCommitteeMember :: L.Credential ColdCommitteeRole StandardCrypto -> Hash CommitteeColdKey
+fromCommitteeMember = \case
+  L.KeyHashObj keyhash -> CommitteeColdKeyHash keyhash
   L.ScriptHashObj _scripthash -> error "TODO script committee members not yet supported"
```
Comment (Jimbo4350) [inline cardano-api/internal/Cardano/Api/Governance/Actions/ProposalProcedure.hs:233]: We should have labelled this with `TODO: Conway` not a big deal though.

**Pull request merged**